qemu/tests/balloon_check.py

```diff
@@ -52,10 +52,15 @@ def get_ballooned_memory(self):
         :rtype: int
         """
         try:
-            output = self.vm.monitor.info("balloon")
-            ballooned_mem = int(re.findall(r"\d+", str(output))[0])
-            if self.vm.monitor.protocol == "qmp":
-                ballooned_mem = ballooned_mem / (1024 ** 2)
+            if self.vm.monitor:
+                output = self.vm.monitor.info("balloon")
+                ballooned_mem = int(re.findall(r"\d+", str(output))[0])
+                if self.vm.monitor.protocol == "qmp":
+                    ballooned_mem = ballooned_mem / (1024 ** 2)
+            else:
+                logging.info('could not get balloon_memory, cause vm.monitor '
+                             'is None')
+                return 0
         except qemu_monitor.MonitorError as emsg:
```

Comment:
Just wondering if this except is still valid, since we already checked vm.monitor in L55.

Reply:
Yes, I think this exception is still valid since `MonitorError` just is generated under the block `if self.vm.monitor: `.

Reply:
we only check if vm.monitor exists in L55, I think we still need this output here

Reply:
Could you pls remind me where MonitorError is raised?

Reply:
If failed to send qmp command, the framework will raise `QMPCmdError` which is defined by `QMPCmdError(MonitorError)`

Reply:
I see, thanks.